File: OjcBlock/OjcBlock/SecondViewController.m
//
//  SecondViewController.m
//  OjcBlock
//
//  Created by 王会洲 on 16/8/5.
//  Copyright © 2016年 王会洲. All rights reserved.
//

#import "SecondViewController.h"

@interface SecondViewController ()

@property (nonatomic, strong) UITextField * inputFiled;

@end

@implementation SecondViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.view.backgroundColor = [UIColor whiteColor];
    self.inputFiled = [[UITextField alloc] initWithFrame:CGRectMake(100, 100, 100, 30)];
    self.inputFiled.borderStyle = UITextBorderStyleRoundedRect;
    [self.view addSubview:self.inputFiled];
    
    
    
    UIButton * playButton = [UIButton buttonWithType:UIButtonTypeCustom];
    playButton.frame = CGRectMake(100, 220, 100, 80);
    [playButton setTitle:@"回调" forState:UIControlStateNormal];
    [playButton setBackgroundColor:[UIColor grayColor]];
    [playButton addTarget:self action:@selector(playVideoBack) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:playButton];
    
    
}
-(void)playVideoBack {
    if (self.changText) {
        self.changText(self.inputFiled.text);
    }
    [self.navigationController popViewControllerAnimated:YES];
}


-(void)setChangText:(changUserName)changText {
    _changText = [changText copy];
}




@end

File: SwiftBlock/SwiftBlock/SwiftBlock/SecondViewController.swift
//
//  SecondViewController.swift
//  SwiftBlock
//
//  Created by 王会洲 on 16/8/5.
//  Copyright © 2016年 王会洲. All rights reserved.
//

import UIKit

typealias changUserName = (String) ->()

class SecondViewController: UIViewController {

    var changText: changUserName?
    
    var changName: UITextField!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        InitView()
    }
    
    
    func InitView()  {
        self.view.backgroundColor = UIColor.whiteColor()
        
        self.changName = UITextField(frame: CGRectMake(100,100,100,50))
        self.changName.borderStyle = .RoundedRect
        self.view.addSubview(self.changName)
        
        
        let pusButton: UIButton = UIButton(type: .Custom)
        pusButton.frame = CGRectMake(100, 220, 100, 80)
        pusButton.backgroundColor = UIColor.redColor()
        self.view.addSubview(pusButton)
        pusButton.addTarget(self, action: #selector(pushClick), forControlEvents: .TouchUpInside)
        
    }
    func pushClick()  {
        changText!(self.changName.text!)
        
        self.navigationController?.popViewControllerAnimated(true)
    }
    
    
    
    func setMyChangeName(tempClose: changUserName)  {
        self.changText = tempClose
    }
    
    

}
